### ISSUE TITLE
nilearn.glm.first_level.first_level_from_bids may incorrectly align models_run_imgs with model_events 

### ISSUE THREAD

**Issue by emaliemcmahon** [nilearn/nilearn#3029]
I used nilearn.glm.first_level.first_level_from_bids to grab my fMRIprep preprocessed data and my event files to run a GLM with nilearn. I noticed that event data frames were not properly aligned with the processed bold images. I think that it is a disparity between the leading zeros filled by dcm2bids but not fmriprep. For instance, the raw nifti files created by dcm2bids are named "run-01", for example (so that is how my event files are named as well), but the fmriprep outputs are instead "run-1". My best guess is that nilearn.glm.first_level.first_level_from_bids is not using natural sorting, which is causing the misalignment. 

Nilearn version: 0.8.0

### Expected behavior
Properly aligned event files to the bold images

### Actual behavior
Misalignment of the event files with the bold images

### Steps and code to reproduce bug
The bug can be reproduced by having event files with leading zeros and preprocessed data without.

**Comment by bthirion** [comment]
I agree that we have to send an informative error msg in that case.
Do you think you can contribute a bug fix by issuing a pull request ?
Best,

**Comment by Remi-Gau** [comment]
Just to note.

The `run` entity in bids is treated as an [index](https://bids-specification.readthedocs.io/en/latest/glossary.html#index-formats) and not a label: 

- https://bids-specification.readthedocs.io/en/latest/appendices/entities.html#run

So even if the file is called `run-01` in one case and `run-1`, it can be a safe bet to assume this is the same run. Most likely one of the filename was not zero padded properly.

Still not sure how nilearn deals with all the different BIDS entities but to be extra cautious, we could just indeed throw an error.

**Comment by bthirion** [comment]
Sorry for my poor knowledge of bids specs. 
As you can see, I am very reluctant to make inference on the provided inputs, when those are ambiguous.
But if other BIDS implementations consistently identify `run-1` and `run-01` then we should do the same.

**Comment by Remi-Gau** [comment]
yup I agree

this kind of BIDS detail is too arcane so I think we should throw an error

will try to see if I can create a test to reproduce the issue

**Comment by Remi-Gau** [comment]
Added a test and fix as part of #3525: it made sense to incorporate the fix with this PR but I could have it on a separate PR that would be merged after #3525 if that's cleaner and easier to review.

See commit https://github.com/nilearn/nilearn/pull/3525/commits/d60e0ad07cb48892503d5be27aca988dd6584831

```python 
def test_first_level_from_bids_bug_3029():
    "Test error when events.tsv is missing for a bold file."
    with InTemporaryDirectory():

        bids_path = create_fake_bids_dataset(n_sub=2,
                                            n_ses=2,
                                            tasks=['main'],
                                            n_runs=[2])
        files_to_rename = Path(bids_path).joinpath('derivatives').glob(
            '**/func/*_task-main_*desc-*')
        for file in files_to_rename:
            new_file = file.parent / file.name.replace('run-0', 'run-')
            file.rename(new_file)

        with pytest.raises(ValueError, 
                           match=".*events.tsv files.*"):
            first_level_from_bids(
                bids_path,
                task_label='main',
                space_label='MNI',
                img_filters=[('desc', 'preproc')],
                verbose=1)
```
